apps/ledger-live-desktop/src/renderer/screens/exchange/Swap2/Form/hooks/useRefreshRates.js

```diff
@@ -0,0 +1,33 @@
+// @flow
+
+import { useRef, useMemo, useEffect } from "react";
+import type { SwapDataType } from "@ledgerhq/live-common/lib/exchange/swap/types";
+import { getRefreshTime } from "~/renderer/screens/exchange/Swap2/utils/getRefreshTime";
+
+const useRefreshRates = (swap: SwapDataType, { pause }: { pause: boolean }) => {
+  const refreshInterval = useRef();
+
+  const refreshTime = useMemo(() => getRefreshTime(swap.rates?.value), [swap.rates?.value]);
+
+  useEffect(() => {
+    clearTimeout(refreshInterval.current);
+
+    refreshInterval.current = setTimeout(() => {
+      console.log("useRefreshRates, calling refetchRates", refreshTime);
+      console.log("useRefreshRates, pause = ", pause);
```

Comment:
Can we remove those console.log?

Reply:
Ah yes, thanks for reminding. Wanted to keep this in until Gregor/Xavier has confirmed it's now working - then will remove